cpp/src/parquet/stream_reader.cc

```diff
@@ -36,6 +36,20 @@ StreamReader::StreamReader(std::unique_ptr<ParquetFileReader> reader)
   NextRowGroup();
 }
 
+int StreamReader::num_columns() const {
+  if (file_metadata_) {
+    return file_metadata_->num_columns();
```

Comment:
Is this only for the null initialization case?  Could you return columns.size() instead to avoid the conditional?

Reply:
Yes the test is only for null initialization.
The test could be avoided by doing what you suggest but I prefer to keep the symmetry with the num_rows() function.

Reply:
might be worth commenting as such.

Reply:
Ok